src/query/api/v1/handler/graphite/find_parser.go

```diff
@@ -24,71 +24,56 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-	"time"
 
 	"github.com/m3db/m3/src/query/errors"
-	"github.com/m3db/m3/src/query/graphite/graphite"
 	graphiteStorage "github.com/m3db/m3/src/query/graphite/storage"
+	"github.com/m3db/m3/src/query/models"
 	"github.com/m3db/m3/src/query/storage"
 	"github.com/m3db/m3/src/query/util/json"
 	"github.com/m3db/m3/src/x/net/http"
 )
 
-func parseFindParamsToQuery(r *http.Request) (
-	*storage.FetchQuery,
+func parseFindParamsToQueries(r *http.Request) (
+	*storage.CompleteTagsQuery,
+	*storage.CompleteTagsQuery,
+	string,
 	*xhttp.ParseError,
 ) {
 	values := r.URL.Query()
-	now := time.Now()
-	fromString, untilString := r.FormValue("from"), r.FormValue("until")
-	if len(fromString) == 0 {
-		fromString = "0"
-	}
-
-	if len(untilString) == 0 {
-		untilString = "now"
+	query := values.Get("query")
+	if query == "" {
+		return nil, nil, "", xhttp.NewParseError(errors.ErrNoQueryFound, http.StatusBadRequest)
 	}
 
-	from, err := graphite.ParseTime(
-		fromString,
-		now,
-		tzOffsetForAbsoluteTime,
-	)
-
+	matchers, err := graphiteStorage.TranslateQueryToMatchersWithTerminator(query)
 	if err != nil {
-		return nil, xhttp.NewParseError(fmt.Errorf("invalid 'from': %s", fromString),
+		return nil, nil, "", xhttp.NewParseError(fmt.Errorf("invalid 'query': %s", query),
 			http.StatusBadRequest)
 	}
 
-	until, err := graphite.ParseTime(
-		untilString,
-		now,
-		tzOffsetForAbsoluteTime,
-	)
+	// NB: Filter will always be the second last term in the matchers, and the
+	// matchers should always have a length of at least 2 (term + terminator)
+	// so this is a sanity check and unexpected in actual execution.
+	filter := [][]byte{matchers[len(matchers)-2].Name}
```

Comment:
Should you check to make sure `len(matchers) > 2`?

Reply:
:| yeah so wrote the comment intending to do the check but ended up forgetting to haha

Reply:
Coolio